mypy/plugins/singledispatch.py

```diff
@@ -0,0 +1,178 @@
+from mypy.plugins.common import add_method_to_class_with_function_type
+from mypy.nodes import ARG_POS, Argument, Block, ClassDef, SymbolTable, TypeInfo, Var
+from mypy.checker import TypeChecker
+from mypy.subtypes import is_subtype
+from mypy.types import (
+    AnyType, CallableType, Instance, NoneType, Overloaded, Type, TypeOfAny, get_proper_type
+)
+from mypy.plugin import CheckerPluginInterface, FunctionContext, MethodContext, MethodSigContext
+from typing import Dict, List, Optional, TypeVar, cast
+from typing_extensions import TypedDict
+
+SingledispatchInfo = TypedDict('SingledispatchInfo', {
+    'fallback': CallableType,
+    # dict of dispatch type to the registered function
+    # dispatch type is stored separately from function because the dispatch type might be different
+    # from the type of the first argument if a type is passed as an argument to register
+    'registered': Dict[Type, CallableType],
+})
+
+RegisterCallableInfo = TypedDict('RegisterCallableInfo', {
+    'register_type': Type,
+    'singledispatch_obj': Instance,
+})
+
+SINGLEDISPATCH_TYPE = 'functools._SingleDispatchCallable'
+
+# key that we use for everything we store in TypeInfo metadata
+METADATA_KEY = 'singledispatch'
+
+
+def get_singledispatch_info(typ: Instance) -> 'SingledispatchInfo':
+    return typ.type.metadata[METADATA_KEY]  # type: ignore
+
+
+T = TypeVar('T')
+
+
+def get_first_arg(args: List[List[T]]) -> Optional[T]:
+    """Get the element that corresponds to the first argument passed to the function"""
+    if args and args[0]:
+        return args[0][0]
+    return None
+
+
+REGISTER_RETURN_CLASS = 'SingleDispatchRegisterCallable'
```

Comment:
Toss an underscore in front

Reply:
Fixed in c0ffb9e4385a5c8af613acd6b5e9d01d47875678.